dashboard/modules/snapshot/tests/test_job_submission.py

```diff
@@ -34,10 +36,24 @@ def _get_snapshot(address: str):
 
 
 @pytest.mark.parametrize("address_suffix", ["", "/"])  # Trailing slash should succeed
+@pytest.mark.parametrize(
+    "set_override_dashboard_url",
+    [
+        None,
+        "https://external_dashboard_url",
+        "https://external_dashboard_url/path1/?query_param1=val1&query_param2=val2",
+        "new_external_dashboard_url",
+    ],
+    indirect=True,
+)
 def test_successful_job_status(
-    ray_start_with_dashboard, disable_aiohttp_cache, enable_test_module, address_suffix
+    set_override_dashboard_url,  # noqa: F811
+    ray_start_with_dashboard,
+    disable_aiohttp_cache,
+    enable_test_module,
+    address_suffix,
 ):
-    address = ray_start_with_dashboard.address_info["webui_url"]
+    address = ray._private.worker._global_node.webui_url
```

Comment:
When an override dashboard url is specified, `ray_context.address_info["webui_url"]` may not be an IP address which can be parsed in `wait_until_server_available`. `ray._private.worker._global_node.webui_url` will contain the local dashboard IP address though. I thought this change was ok because this address is being passed in to the public `JobSubmissionClient` API and is not being formed internally.

Reply:
I see, so in general, if we use:
- `ray_context.address_info["webui_url"]`: this is the overriden URL if set. And not guaranteed to be queryable with APIs like in state API/jobs
- `ray._private.worker._global_node.webui_url`: this is the old `<host>:<port>` url, that refers to the address of the dashboard (if present) directly. 

Is this right? If so, would you also documenting this invariance somewhere in the code? Maybe when `ray_context` is first created so users of this would know the subtle differences here.